### ISSUE TITLE
Ignatius Out of Memory

### ISSUE THREAD

**Issue by zongh5a** [BaowenZ/RaDe-GS#83]
When training the model on the **Ignatius** scene, the training process hangs and freezes after approximately 5600 iterations. This appears to be caused by VRAM exhaustion (Out of Memory). 

### Hardware Environment
- **GPU:** NVIDIA RTX 5090 (32GB VRAM)

### Proposed Fix / Workaround
I managed to fix this issue by explicitly clearing the CUDA cache at the end of the `densify_and_prune` function. Adding `torch.cuda.empty_cache()` prevents the memory from blowing up during the densification and pruning steps.

Here is the modified code snippet:

```python
    def densify_and_prune(self, max_grad, min_opacity, extent, max_screen_size):
        grads = self.xyz_gradient_accum / self.denom
        grads[grads.isnan()] = 0.0

        grads_abs = self.xyz_gradient_accum_abs / self.denom
        grads_abs[grads_abs.isnan()] = 0.0
        ratio = (torch.norm(grads, dim=-1) >= max_grad).float().mean()
        Q = torch.quantile(grads_abs.reshape(-1), 1 - ratio)

        before = self._xyz.shape[0]
        self.densify_and_clone(grads, max_grad, grads_abs, Q, extent)
        clone = self._xyz.shape[0]

        self.densify_and_split(grads, max_grad, grads_abs, Q, extent)
        split = self._xyz.shape[0]

        prune_mask = (self.get_opacity < min_opacity).squeeze()
        # if max_screen_size:
        #     big_points_vs = self.max_radii2D > max_screen_size
        #     big_points_ws = self.get_scaling.max(dim=1).values > 0.1 * extent
        #     prune_mask = torch.logical_or(torch.logical_or(prune_mask, big_points_vs), big_points_ws)
        self.prune_points(prune_mask)
        prune = self._xyz.shape[0]

        # Fix: Free up GPU memory to prevent freezing
        torch.cuda.empty_cache()

        return clone - before, split - clone, split - prune
